### PR TITLE
Run kudu on mono

### PR DESCRIPTION
**Pull request by shrimpy**
- Run kudu on mono
- Solve issue that deployment lock not working on kudu.exe
- Update PathUtility to aware of linux
- Grant full permission to post receive hook
- Hide unsupported Kudu feature from Linux enviroment.

### CODE REVIEW

File: Common/Constants.cs [hunk 1]
```diff
@@ -22,7 +23,7 @@ public static class Constants
         public const string SiteFolder = @"site";
         public const string LogFilesPath = @"LogFiles";
         public const string ApplicationLogFilesDirectory = "Application";
-        public const string TracePath = LogFilesPath + @"\kudu\trace";
+        public readonly static string TracePath = Path.Combine(LogFilesPath, "kudu", "trace");
```
Comment (davidebbo) [inline Common/Constants.cs]: Why this change? Having a true const seems preferable.

Comment (shrimpy) [inline Common/Constants.cs]: When running on mono, it doesn`t like the path separator '\', and we end up creating a folder name with "LogFiles\kudu\trace" instead of nested folders. using "Path.Combine" it will use the right separator when running on different environment.
